src/unstract/sdk/llm.py

```diff
@@ -120,10 +127,75 @@ def _get_max_tokens(self, reserved_for_output: int = 0) -> int:
         """
         return self.MAX_TOKENS - reserved_for_output
 
-    @deprecated("Use the new class LLM")
+    def set_max_tokens(self, max_tokens: int) -> None:
+        """Set the maximum number of tokens that can be used for the LLM.
+
+        Args:
+            max_tokens (int): The number of tokens to be used at the maximum
+
+            Returns:
+                None
+        """
+        self._llm_instance.max_tokens = max_tokens
+
+    def get_class_name(self) -> str:
+        """Gets the class name of the Llama Index LLM.
+
+        Args:
+            NA
+
+            Returns:
+                Class name
+        """
+        return self._llm_instance.class_name()
+
+    @deprecated("Use LLM instead of ToolLLM")
     def get_llm(self, adapter_instance_id: Optional[str] = None) -> LlamaIndexLLM:
+        if not self._llm_instance:
+            self._adapter_instance_id = adapter_instance_id
+            self._initialise()
         return self._llm_instance
 
+    @classmethod
+    @deprecated("Instantiate LLM and call complete() instead")
+    def run_completion(
+        cls,
+        llm: LlamaIndexLLM,
+        platform_api_key: str,
+        prompt: str,
+        retries: int = 3,
+        **kwargs: Any,
+    ) -> Optional[dict[str, Any]]:
+        # Setup callback manager to collect Usage stats
+        CallbackManager.set_callback_manager(
+            platform_api_key=platform_api_key, llm=llm, **kwargs
+        )
+        # Removing specific keys from kwargs
+        new_kwargs = kwargs.copy()
+        for key in [
+            "workflow_id",
+            "execution_id",
+            "adapter_instance_id",
+            "run_id",
+        ]:
+            new_kwargs.pop(key, None)
+
+        try:
+            response: CompletionResponse = llm.complete(prompt, **new_kwargs)
+            match = cls.json_regex.search(response.text)
+            if match:
+                response.text = match.group(0)
+            return {"response": response}
+        # TODO: Handle for all LLM providers
+        except OpenAIAPIError as e:
```

Comment:
@gaya3-zipstack Looks like this error is specific to OpenAI. can this be handled in the adapter level?

Reply:
Yes @harini-venkataraman - it should ideally be at the adapter level. I had to add this handling here when we still had some of the old changes. If this error is handled at the adapter level code for an LLM call we can remove this block @gaya3-zipstack

Reply:
@harini-venkataraman @chandrasekharan-zipstack In order to take it to adapter level, we need to spport all these in adapter leve. Will reach there at some point ...

Reply:
Okay - until its done, let this remain until then